httpx/_utils.py

```diff
@@ -529,3 +533,20 @@ def __eq__(self, other: typing.Any) -> bool:
 
 def warn_deprecated(message: str) -> None:  # pragma: nocover
     warnings.warn(message, DeprecationWarning, stacklevel=2)
+
+
+async def sleep(seconds: float) -> None:
+    library = sniffio.current_async_library()
+    if library == "trio":
+        import trio
+
+        await trio.sleep(seconds)
+    else:
+        assert library == "asyncio"
+        await asyncio.sleep(seconds)
```

Comment:
I guess we may as well also include curio here, in anticipation of the incoming support in `httpcore` - https://github.com/encode/httpcore/pull/168

We can always just `pragma: nocover` it for the moment, right?